src/daft-session/src/python.rs

```diff
@@ -95,6 +95,12 @@ impl PySession {
     }
 }
 
+impl From<&PySession> for Session {
+    fn from(sess: &PySession) -> Self {
+        sess.0.clone()
+    }
+}
+
```

Comment:
super minor nit: but why not `impl From<PySession>` instead of `&PySession`. this just hides the clone. 

```rs
let sess: &Pysession;
let sess: Session = sess.clone().into(); // the clone is explicit
// vs
let sess: Session = sess.into(); // the clone is implicit
```